src/server/process_request.c

```diff
@@ -287,19 +288,15 @@ req_register_sched(conn_t *conn, struct batch_request *preq)
 		rc = PBSE_BADHOST;
 		goto rerr;
 	}
-	if (sched->sc_primary_conn != -1 || sched->sc_secondary_conn != -1) {
+	if (sched->sc_primary_conn != -1 && sched->sc_secondary_conn != -1) {
 		rc = PBSE_SCHEDCONNECTED;
 		goto rerr;
 	}
-	if (strcmp(preq->rq_ind.rq_rsched.rq_conn_type, "primary") == 0) {
+	if (sched->sc_tmp_primary_conn == -1 && sched->sc_primary_conn == -1) {
```

Comment:
No, we are not longer sending 'conn_type' anymore, as it will allow us (in future) to fulfill use cases like your 2nd pbs_est use case, where will have only connection (not 2 connections like in sched) and still mark connection as super privileged connection. (just FYI, for now we are not saying anything for other use cases in design or not making any code/func names generic, those will be taken care by next project (aka - non-root daemons project - which is dependent on this changes), as will have better picture there)

About, out of sync, it shouldn't happen as server will get notified for close connection if sched crashes or restarts just after making first connection (aka primary connection), as its tcp connection and if process dies then that connection will be closed and other side of connection (aka server in this case) will get notified. And once server gets notified it should remove socket from sched obj.